### DISCUSSION TITLE
Resetting a single field to default values isn't fun when using useFormContext

### DISCUSSION THREAD

**Discussion by 06kellyjac** [react-hook-form/react-hook-form#3861] [Q&A] [answered]
Im using useFormContext and enjoying it a lot but I want to have individual reset buttons per field.
Resetting to the default values is a pain because I have to pass through the defaultValues as useFormContext doesn't have a way to fetch it and reset doesn't let you just reset a key and let it handle the value.
I could set up something using context or zustand to get the default values but then ill also need to handle nested values myself too e.g. "something.otherthing

Thanks for reading and thanks for making this such a great library

**Comment by bluebill1049** [comment, marked as answer]
why not export defaultValue as an object?

**Reply by 06kellyjac** [reply]
I might not be understanding correctly but here's a bit more detail:

I want composable form blocks that are easy to use without passing props where possible. And I wanted to have a `<ResetButton name="whatever"/>` component where as long as I provide the `name` it'd reset that value.

If i pass the `defaultValues` in using props I'd have to pass them to my form blocks then to my input field components and from there into my `ResetButton`s

---

I have now got a context where I can set the `defaultValues` (if they exist, it could be a creation form rather than an edit form which would have exiting `defaultValues`)
I have a function that can pull out a value from `defaultValues` using the name e.g. `"technology.whatever"` but then I'd also need a different function to re-construct the object not just with `reset({...getValues(), [name]: theDefaultValue})` but having (in this case) `reset({...getValues(), technology: { whatever: theDefaultValue }})`.
To avoid that I'm just using `setValue(name, theDefaultValue, { shouldValidate: true, shouldDirty: false })`.

Generally this workflow is working quite well and is pretty comfortable.
I can use blocks of form inputs across my "create" wizard form and my long single "edit" form without passing loads of variables down. It would be nice if there was a built in way to do things.

**Reply by bluebill1049** [reply]
I see, you are just after the `defaultValues` throughout the context. I think what you doing right now it's the correct approach.

**Reply by 06kellyjac** [reply]
Thanks for your time. It was helpful to put it into words and check I wasnt doing anything crazy or if there was something easy I missed
